include/concepts/concepts.hpp

```diff
@@ -1043,9 +1025,43 @@ namespace concepts
         // Comparison concepts
         ////////////////////////////////////////////////////////////////////////////////////////////
 
+        template<typename T>
+        CPP_concept_bool boolean_testable_impl_ = convertible_to<T, bool>;
+
+        template<typename T>
+        CPP_concept_fragment(boolean_testable_frag_,
+            requires(T && t) //
+            (
+                !(T&&) t,
+                concepts::requires_<boolean_testable_impl_<decltype(!(T&&) t)>>
+            ) &&
+            boolean_testable_impl_<T>
+        );
+        template<typename T>
+        CPP_concept_bool boolean_testable_ =
+            CPP_fragment(boolean_testable_frag_, T);
+
+        CPP_DIAGNOSTIC_PUSH
+        CPP_DIAGNOSTIC_IGNORE_FLOAT_EQUAL
+
+        template<typename T, typename U>
+        CPP_concept_fragment(weakly_equality_comparable_with_frag_,
+            requires(detail::as_cref_t<T> t, detail::as_cref_t<U> u) //
+            (
+                concepts::requires_<boolean_testable_<decltype(t == u)>>,
+                concepts::requires_<boolean_testable_<decltype(t != u)>>,
+                concepts::requires_<boolean_testable_<decltype(u == t)>>,
+                concepts::requires_<boolean_testable_<decltype(u != t)>>
+            ));
+        template<typename T, typename U>
+        CPP_concept_bool weakly_equality_comparable_with_ =
+            CPP_fragment(weakly_equality_comparable_with_frag_, T, U);
```

Comment:
I thought you were going to keep this in a detail namespace. I don't want it in the `ranges::` namespace. I apologize if I was confused about what you asking earlier.

Reply:
Okay I kept the namespace `concepts::detail` by closing the `concepts::defs` namespace and reopening it later. I couldn't just add a `concepts::defs::detail` namespace, since it would clash with `concepts::detail.`